backend/addcorpus/python_corpora/corpus.py

```diff
@@ -352,49 +354,68 @@ class FieldDefinition(Field):
+        csv_core: Whether the field is pre-selected for CSV downloads of search results.
+        search_field_core: Whether the field is immediately shown in field selection for
+            the search query. If `False`, the field is only shown when the user selects
+            "show all fields".
+        visualizations: Visualisations that are enabled for this field. Options:
+            resultscount, termfrequency, wordcloud, ngram. For date fields and
+            categorical/ordinal fields (usually keyword type), you can use
+            `['resultscount', 'termfrequency']`. For text fields, you can use
+            `['wordcloud', 'ngram']`. However, the ngram visualisation also requires that
+            the corpus has a date field.
+        visualization_sort: If the visualisations include resultscount or termfrequency
+            and the field is not a date field, this determines how the histogram is
+            sorted. Options are `'value'` (sort from most to least frequent) or `'key'`
+            (sort alphabetically).
+        es_mapping: The mapping of the field in Elasticsearch. It's recommended to use one
+            of the functions in `addcorpus.es_mappings` to construct this.
+        language: The language of the field's content. Can be `None`, an IETF tag, or
+            `"dynamic"`. None means the language is unknown or NA. Dynamic means the
+            `language_field` of the corpus specifies the IETF tag for this field's
+            language per document.
+        search_filter: Configuration of the search filter used for the field (if any).
+            Should be `Filter` instance.
+        extractor: Configuration to extract the field's data from source documents. Should
+            be an `Extractor` instance.
+        sortable: Whether this field is shown as an option to sort search results.
+        searchable: Whether this field is shown in the selection for search fields.
+        downloadable: Whether this field may be included when downloading results.
+        required: Whether this field is required during source extraction. Note that not
+            all Reader subclasses currently support this.
     '''
 
     def __init__(self,
-                 name=None,
-                 display_name=None,
-                 display_type=None,
-                 description='',
-                 indexed=True,
-                 hidden=False,
-                 results_overview=False,
-                 csv_core=False,
-                 search_field_core=False,
-                 visualizations=[],
-                 visualization_sort=None,
-                 es_mapping={'type': 'text'},
-                 language=None,
-                 search_filter=None,
-                 extractor=extract.Constant(None),
-                 sortable=None,
-                 searchable=None,
-                 downloadable=True,
-                 required=False,
+                 name: str = None,
```

Comment:
```suggestion
                 name: Optional[str] = None,
```

Reply:
I wasn't sure about this. `None` is not actually valid input here.